java-checks/src/main/java/org/sonar/java/checks/StaticMemberAccessCheck.java

```diff
@@ -56,7 +56,9 @@ public void visitNode(Tree tree) {
     if (symbol.isStatic()  && !isListOrSetOf(mse)) {
       ExpressionTree expression = mse.expression();
       Type staticType = symbol.owner().type();
-      if (!expression.symbolType().erasure().equals(staticType.erasure())) {
+      Type expressionType = expression.symbolType();
+      if (!staticType.isUnknown() && !expressionType.isUnknown()
```

Comment:
In practice, is `expressionType.isUnknown()` ever true?

Reply:
Indeed, I did not manage to come up with a use-case that cover this case, but I decided to keep it for defensive programming.